lib/rex/post/meterpreter/extensions/stdapi/sys/registry.rb

```diff
@@ -215,10 +215,17 @@ def Registry.set_value(hkey, name, type, data)
     request.add_tlv(TLV_TYPE_VALUE_NAME, name)
     request.add_tlv(TLV_TYPE_VALUE_TYPE, type)
 
-    if type == REG_SZ || type == REG_MULTI_SZ
-      data += "\x00"
-    elsif (type == REG_DWORD)
-      data = [ data.to_i ].pack("V")
+    case type
+    when REG_DWORD
+      data = [data.to_i].pack('L<')
+    when REG_EXPAND_SZ
+      data += "\x00".b
+    when REG_MULTI_SZ
+      data = data.join("\x00".b) + "\x00\x00".b
+    when REG_QWORD
+      data = [data.to_i].pack('Q<')
+    when REG_SZ
+      data += "\x00".b
     end
```

Comment:
This and `set_value_direct` look to share pattern should we consolidate that logic a private helper?